### PR TITLE
Add kill assist

### PR DESCRIPTION
**Pull request by Anthony-De**
Add's a feature that gives points to players who do more than 50% damage and don't get the kill.

### CODE REVIEW

File: mods/ctf/ctf/teams.lua [hunk 1]
```diff
@@ -437,6 +437,18 @@ minetest.register_on_joinplayer(function(player)
 	end
 end)
 
+function ctf.clearAssists(playerName)
```
Comment (Panquesito7) [inline mods/ctf/ctf/teams.lua:440]: Would be better if you name this function as  `ctf.clear_assists`.

Comment (Panquesito7) [inline mods/ctf/ctf/teams.lua:440]: You would need to change all `playerName` instances to `player`.
```suggestion
function ctf.clearAssists(player)
```

Comment (Anthony-De) [reply mods/ctf/ctf/teams.lua]: I have applied this.

Comment (Anthony-De) [reply mods/ctf/ctf/teams.lua]: I have applied this.

**Review by moreavy — COMMENTED** [review]
I don't think that adding kill assist is a good idea. Reasons:-
1) There are noobs who give just 2.5 points when someone kills them. If 3-4 people attack him at once, they will get really less score.
2) People will get just more kills by not killing a person and hitting him/her 2 time.
3) **Steal Kill** thing will increase a lot.
4) People's k/d will increase too much and noobs not eligible for *pro* will get pro chest.
5) If a person gives 100 points when he/she dies, he/she will give only 33.3 points if 3 people kill him together.

**Comment by JostP** [comment]
> I don't think that adding kill assist is a good idea. Reasons:-
> 
>     1. There are noobs who give just 2.5 points when someone kills them. If 3-4 people attack him at once, they will get really less score.
> 
>     2. People will get just more kills by not killing a person and hitting him/her 2 time.
> 
>     3. **Steal Kill** thing will increase a lot.
> 
>     4. People's k/d will increase too much and noobs not eligible for _pro_ will get pro chest.
> 
>     5. If a person gives 100 points when he/she dies, he/she will give only 33.3 points if 3 people kill him/her together.

I didn't check how does the code work.
- My opinion is that kill assist **should not count as kill**. But it should be displayed on kill history list, something like:

<killer_name> + <assistant_name> :crossed_swords: <death_player>

- Only people who give more than 50% of total health damage should get assist points
- The algorithm should include what happens when victim heals.

**Comment by moreavy** [comment]
@JostP What if 3-4 people together kill a person?

**Comment by JostP** [comment]
> @JostP What if 3-4 people together kill a person?

Then if 3 people give 25% of damage and killer kills victim, only killer get points because all other assistants did less than 50% of total health damage.

**Comment by moreavy** [comment]
No I didn't

**Comment by moreavy** [comment]
But Kill Assist should not get point.. That will really not be good.

**Comment by TSafa-23** [comment]
I like the idea of kill assists as it encourages teamwork, and to sniper I think that that limit would probably be better as a 33% limit (imo it's an assist, not killing half and half)

**Comment by Anthony-De** [comment]
For anyone who hasn't checked the code, 
A kill assist doesn't count as a kill. 
Nor does it subtract from the actual killers' score.

Example:
PlayerA, PlayerB, and PlayerC are attacking PlayerD.
PlayerA does 90% of damage to PlayerD.
PlayerD kills PlayerA.
PlayerE heals PlayerD.
PlayerB and PlayerC do 50% damage each.
PlayerC does the final stroke.

PlayerD's death reward is 40pts

PlayerC gets 40pts and the kill
PlayerA gets 30pts because the assist caps at 75%
PlayerB gets 20pts

@moreavy

**Comment by Anthony-De** [comment]
> @JostP 
> <killer_name> + <assistant_name> ⚔️ <dead_player>

I have considered this (see above example) there would be too many names to list if there were more than 2 people involved.

**Comment by xenonca** [comment]
As a possible reference you might want to look at #184

**Comment by moreavy** [comment]
Code crashed
image

**Comment by moreavy** [comment]
_Suggestion_: add Capture Assist with the kill assist.

**Comment by Panquesito7** [comment]
> _Suggestion_: add Capture Assist with the kill assist.

This should be made in another PR.

**Comment by Anthony-De** [comment]
Once this is merged, I will start on the Capture assist feature.

**Review by Panquesito7 — APPROVED** [review]
Tested; works. 👍

**Comment by Anthony-De** [comment]
@dependabot rebase

**Pull request merged**